google-cloud-datastore/src/main/java/com/google/cloud/datastore/RecordQuery.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.datastore;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.datastore.Query.ResultType;
+
+/** An internal marker interface to represent {@link Query} that returns the entity records. */
+@InternalApi
+public interface RecordQuery<V> {
+
+  ResultType<V> getType();
```

Comment:
is this also meant to be internal?

Reply:
Yup, it was package private earlier, so an internal function here also. just fixed in https://github.com/googleapis/java-datastore/pull/875/commits/aa834031e36be75c489525fca1d4734edf374ba2.